Makefile

```diff
@@ -15,101 +15,83 @@
-	cp -R $(STAGING_BASE)* $(DESTDIR)$(PREFIX)
-	dpkg-deb -Zxz --root-owner-group --build $(DESTDIR) $(MACHINE_EMULATOR_TOOLS_DEB)
-
-$(LINUX_SOURCES_FILEPATH):
-	T=`mktemp` && \
-	    mkdir -p `dirname $@` &&  \
-	    wget $(LINUX_SOURCES_URLPATH) -O $$T && \
-	    mv $$T $@ || (rm $$T && false)
+$(TOOLS_DEB) deb: build
 
-$(RNDADDENTROPY_FILEPATH):
-	T=`mktemp` && \
-	    mkdir -p `dirname $@` &&  \
-	    wget $(RNDADDENTROPY_URLPATH) -O $$T && \
-	    mv $$T $@ || (rm $$T && false)
+control:
+	@sed 's|ARG_VERSION|$(VERSION)|g' control.template > control
 
-shasumfile: $(SHASUMFILES)
-	shasum -a 256 $^ > $@
 
-checksum: $(SHASUMFILES)
-	@shasum -ca 256 shasumfile
+$(TOOLS_ROOTFS) fs: $(TOOLS_DEB)
+	@docker buildx build --platform=linux/riscv64 \
+	  --build-arg TOOLS=$(TOOLS_DEB) \
+	  --output type=tar,dest=rootfs.tar \
+	  --file fs/Dockerfile \
+	  . && \
+	bsdtar -cf rootfs.gnutar --format=gnutar @rootfs.tar && \
+	xgenext2fs -fzB 4096 -i 4096 -a rootfs.gnutar -L rootfs $(TOOLS_ROOTFS) && \
+	rm -f rootfs.gnutar rootfs.tar
 
 env:
-	@echo MACHINE_EMULATOR_TOOLS_TAR_GZ=$(MACHINE_EMULATOR_TOOLS_TAR_GZ)
-	@echo MACHINE_EMULATOR_TOOLS_DEB=$(MACHINE_EMULATOR_TOOLS_DEB)
-	@echo MACHINE_EMULATOR_TOOLS_VERSION=$(MACHINE_EMULATOR_TOOLS_VERSION)
-	@echo LINUX_SOURCES_VERSION=$(LINUX_SOURCES_VERSION)
-	@echo LINUX_SOURCES_FILEPATH=$(LINUX_SOURCES_FILEPATH)
-	@echo RNDADDENTROPY_VERSION=$(RNDADDENTROPY_VERSION)
-	@echo RNDADDENTROPY_FILEPATH=$(RNDADDENTROPY_FILEPATH)
+	@echo TOOLS_DEB=$(TOOLS_DEB)
+	@echo TOOLS_ROOTFS=$(TOOLS_ROOTFS)
+	@echo TOOLS_IMAGE=$(TOOLS_IMAGE)
+	@echo IMAGE_KERNEL_VERSION=$(IMAGE_KERNEL_VERSION)
+	@echo LINUX_VERSION=$(LINUX_VERSION)
+	@echo LINUX_HEADERS_URLPATH=$(LINUX_HEADERS_URLPATH)
 
 setup:
-	docker run --privileged --rm  linuxkit/binfmt:bebbae0c1100ebf7bf2ad4dfb9dfd719cf0ef132
+	@docker run --privileged --rm  linuxkit/binfmt:bebbae0c1100ebf7bf2ad4dfb9dfd719cf0ef132
 
 setup-required:
 	@echo 'riscv64 buildx setup required:' `docker buildx ls | grep -q riscv64 && echo no || echo yes`
 
 help:
 	@echo 'available commands:'
+	@echo '  deb             - build machine-emulator-tools .deb package'
+	@echo '  fs              - build rootfs.ext2'
 	@echo '  setup           - setup riscv64 buildx'
 	@echo '  setup-required  - check if riscv64 buildx setup is required'
-	@echo '+ tgz             - create "$(MACHINE_EMULATOR_TOOLS_TAR_GZ)" (default)'
 	@echo '  help            - list makefile commands'
-	@echo '  checksum        - verify downloaded files'
-	@echo '  shasumfile      - recreate shasumfile of downloaded files'
 	@echo '  env             - print useful Makefile variables as a KEY=VALUE list'
 	@echo '  clean           - remove the generated artifacts'
-	@echo '  distclean       - clean and remove dependencies'
 
-clean:
-	rm -f $(MACHINE_EMULATOR_TOOLS_TAR_GZ) $(MACHINE_EMULATOR_TOOLS_DEB)
+clean-image:
+	@(docker rmi $(TOOLS_IMAGE) > /dev/null 2>&1 || true)
 
-distclean: clean
-	rm -rf dep
+distclean clean: clean-image
```

Comment:
Will `clean` require rebuilding the docker images? I don't like it if that is the case.